php/pro/BaseCache.php

```diff
@@ -53,4 +53,19 @@ public function offsetUnset($index) : void {
     public function __toString() {
         return print_r($this->deque, true);
     }
+
+    public function convert_from_snake_to_camel_case($input)
+    {
+        return lcfirst(str_replace(' ', '', ucwords(str_replace('_', ' ', $input))));
+    }
+
+    // tmp support for type bcz of transpiler snake_case
+    public function __call($name, $arguments)
+    {
+        $camelCased = $this->convert_from_snake_to_camel_case($name);
+        if (method_exists($this, $camelCased)) {
+            return call_user_func_array(array($this, $camelCased), $arguments);
+        }
+        trigger_error('Call to undefined method '.__CLASS__.'::'.$name.'()', E_USER_ERROR);
```

Comment:
@ttodua can't we just add 

```
    public function get_limit() : int {
        return $this->getLimit()
    }
```
instead?

Reply:
I though automatic way would have been better, because anytime in future there might be amended function-names, added functions or renamed things. why to hardcode?

Reply:
but ok i'll do that as you said

Reply:
> be amended function-names, added functions or renamed things. why to hardcode
because if the transpiler transpiles getLimit to something weird is much easier to spot and besides that it simplifies the code